src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/ChunkVisitor.cs

```diff
@@ -1,14 +1,14 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Generic;
+using Microsoft.Framework.Internal;
 
 namespace Microsoft.AspNet.Razor.Generator.Compiler
 {
     public abstract class ChunkVisitor<T> : IChunkVisitor where T : CodeWriter
     {
-        public ChunkVisitor(T writer, CodeBuilderContext context)
+        public ChunkVisitor([NotNull] T writer, [NotNull] CodeBuilderContext context)
```

Comment:
Unrelated - could you change the type param to `TWriter` whilst you're touching this file?